nilearn/datasets/neurovault.py

```diff
@@ -1407,20 +1419,20 @@ def _download_collection(collection, download_params):
     """
     if collection is None:
         return None
+
     collection = _remove_none_strings(collection)
     collection_id = collection["id"]
     collection_name = f"collection_{collection_id}"
-    collection_dir = os.path.join(
-        download_params["nv_data_dir"], collection_name
-    )
+    collection_dir = Path(download_params["nv_data_dir"]) / collection_name
     collection["relative_path"] = collection_name
-    collection["absolute_path"] = collection_dir
-    if not os.path.isdir(collection_dir):
-        os.makedirs(collection_dir)
-    metadata_file_path = os.path.join(
-        collection_dir, "collection_metadata.json"
-    )
+    collection["absolute_path"] = str(collection_dir.absolute())
```

Comment:
Am wondering if we have to cast to string here given this is a private function.
Does a lot of downstream things ion the code break if we stick to Path?

Reply:
The tests do not pass if I use Path. 

I've checked if it will be straightforward. But I do not think so.  

It might be easier to come back to this after all PTH errors in https://github.com/nilearn/nilearn/issues/4327 are solved. 

I suggest to create an issue for that to come back later.

Reply:
I agree let's flag that in another issue to tackle later